packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -1174,9 +1174,8 @@ export function makeClientIdInDEV(warnOnAccessInDEV: () => void): OpaqueIDType {
   };
 }
 
-let serverId: number = 0;
-export function makeServerId(): OpaqueIDType {
-  return 'R:' + (serverId++).toString(36);
+export function makeServerId(prefix: ?string, serverId: number): OpaqueIDType {
```

Comment:
Why is `prefix` optional? We might as well always pass a string from the outside.

Reply:
We want to minimize the number of bytes we send, so we don't want to append a `prefix` unless necessary (ie for apps that render multiple subtrees on the server and then stitch them together).

Reply:
Flow types suggest "prefix" is optional but "serverId" is required. Is this the intent?

Putting an optional param before a required param seems wrong.